test/WebJobs.Script.Tests.Integration/TestFunctionHost.cs

```diff
@@ -158,6 +158,14 @@ public class TestFunctionHost : IDisposable
             HttpClient = _testServer.CreateClient();
             HttpClient.Timeout = TimeSpan.FromMinutes(5);
 
+            var environment = _testServer.Services.GetService<IEnvironment>();
+            if (environment.IsAppService())
+            {
+                // host is configured to simulate an AppService environment
+                // all normal requests will go through the Antares FrontEnd and receive this header
+                HttpClient.DefaultRequestHeaders.Add(ScriptConstants.AntaresLogIdHeaderName, "xyz");
```

Comment:
Many tests were previously being granted internal access on requests because the test requests weren't sending this header. Some of those tests are testing the auth pipeline so I'm fixing this and making things explicit/consistent. With this change, by if the TestHost is configured to be running in AppService, requests sent with this client will be external (will have this header). For the handful of tests that are testing internal requests, they'll use a client that doesn't add this header.